make-pangenome-sketches.py

```diff
@@ -21,52 +23,88 @@ def main():
     )
     p.add_argument('--scaled', default=1000, type=int)
     p.add_argument('-k', '--ksize', default=31, type=int)
-    p.add_argument('-o', '--output', required=True)
+    p.add_argument('-o', '--output', required=True,
+                    help='Define a filename for the pangenome signatures (.zip preferred). \n'
+                         'A CSV file with the same name will also be generated to contain the lineage rank, genome name, and hash count.')
     p.add_argument('-r', '--rank', default='species')
+    p.add_argument('-a', '--abund', action='store_true', help='Enable abundance tracking of hashes across rank selection.')
     args = p.parse_args()
 
     print(f"loading taxonomies from {args.taxonomy_file}")
     taxdb = sourmash.tax.tax_utils.MultiLineageDB.load(args.taxonomy_file)
     print(f"found {len(taxdb)} identifiers in taxdb.")
 
-    revtax_d = {}
     ident_d = {}
+    revtax_d = {}
+    accum = defaultdict(dict)
+    if args.abund:
+        counts = {}
+    csv_file = check_csv(args.output)
 
+    # Load the database
     for filename in args.sketches:
         print(f"loading file {filename} as index => manifest")
         db = sourmash_args.load_file_as_index(filename)
         db = db.select(ksize=args.ksize)
         mf = db.manifest
         assert mf, "no matching sketches for given ksize!?"
 
-        # load and merge
+        chunk = []
+        # Work on a single signature at a time across the database
         for n, ss in enumerate(db.signatures()):
             if n and n % 1000 == 0:
                 print(f'...{n} - loading')
-            name = ss.name
-            md5sum = ss.md5sum()
 
+            name = ss.name
             ident = tax_utils.get_ident(name)
-
+            
             # grab relevant lineage name
             lineage_tup = taxdb[ident]
             lineage_tup = tax_utils.RankLineageInfo(lineage=lineage_tup)
             lineage_pair = lineage_tup.lineage_at_rank(args.rank)
             lineage_name = lineage_pair[-1].name
 
             # pick an ident to represent this set of pangenome sketches
-            ident_d[lineage_name] = ident # ok if overwrite...
+            ident_d[lineage_name] = ident 
+
+            # Accumulate the count within lineage names if `--abund` in cli
```

Comment:
it looks to me like you're tracking counts separately in a Counter, and then using `set_abunds` later on to update the `MinHash` object with - is there a reason not to track the counts within the `MinHash` object to begin with? I don't immediately see where else you are using counts for something.

Reply:
Using `collections.counter` creates a subclass dict `Counter` object, that is why [L119](https://github.com/ctb/2022-database-covers/blob/de0cc2602f17382a5cff646c7870c2ca896f284c/make-pangenome-sketches.py#L119) converts it back into a standard dict object. I was not sure if this would be handled well in sourmash API.

This was also my first time jumping into the sourmash API and I was not entirely sure how to achieve what you describe, I did try but decided that I would start with something that I implicitly understood.

Reply:
gotcha! sounds good. let me know when I should take a look at this again and maybe I can offer some guidance.

Also, any concerns about moving this over to github.com/dib-lab/?

Reply:
This comment thread is where I think I need the most guidance. I would like to know more about sourmash internals. Our meeting discussing manifests and zip files helped me out with this, surprisingly.

No concerns. Just honor.

Reply:
ok, I dug a bit into https://github.com/sourmash-bio/sourmash/blob/latest/src/sourmash/minhash.py#L4 and the following should work -

* change the code in `if mh is None` to say:

```
mh = ss.minhash.copy_and_clear()
mh.track_abundance = True
mh += ss.minhash
revtax_d[lineage_name] = mh
```

and see if that works ;). Then you should be able to get rid of most of your Counter stuff, because the minhash object will be tracking abunds.

It may not work b/c you may not be able to add a flat MinHash to an abund MinHash, tho.